nomad/structs/structs.go

```diff
@@ -7340,6 +7392,36 @@ func (tg *TaskGroup) GoString() string {
 	return fmt.Sprintf("*%#v", *tg)
 }
 
+// GetDisconnectLostTimeout is a helper meant to simplify the future depracation of
+// MaxClientDisconnect in favor of Disconnect.LostAfter
+// introduced in 1.8.0.
+func (tg *TaskGroup) GetDisconnectLostTimeout() time.Duration {
+	if tg.MaxClientDisconnect != nil {
+		return *tg.MaxClientDisconnect
+	}
```

Comment:
Maybe I'm confusing it with another field because I can't find any docs about it, but is `0` a valid (and special) value for `max_client_disconnect`?

I have a vague memory of a bug around this area that was caused by mishandling a `0` value 😅

Reply:
I think the one you are thinking about is stop_after_client_disconnect